### PR TITLE
smoothly switch to a XDG layout for global files

### PR DESCRIPTION
**Pull request by doudou**
The code should pick a XDG location for new files/folders, but keep
the current locations when they exist.

### CODE REVIEW

File: lib/autoproj/ops/install.rb [hunk 1]
```diff
@@ -52,8 +52,16 @@ def initialize(root_dir)
                 @ruby_executable = config['ruby_executable']
                 @local = false
 
+                xdg_default_gem_path = File.join(Dir.home, '.local', 'share', 'autoproj', 'gems')
```
Comment (g-arjones) [inline lib/autoproj/ops/install.rb]: Why isn't this XDG_DATA_HOME instead? Same for the bootstrap/install scripts..

Ref: https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html

Comment (doudou) [reply lib/autoproj/ops/install.rb]: This is XDG_DATA_HOME.

Unfortunately, the xdg gem actually does not handle the environment variables - it only handles the default XDG folders in Dir.home. Moreover, install.rb should be dependency-free (it ends up included in the install/bootstrap script) and therefore cannot use the xdg gem.

Comment (doudou) [reply lib/autoproj/ops/install.rb]: > Unfortunately, the xdg gem actually does not handle the environment variables

Well .. this is actually not true.

Comment (doudou) [reply lib/autoproj/ops/install.rb]: ... and I therefore added support for the XDG_DATA_HOME env var.

**Comment by g-arjones** [comment]
Nice.. Looks good to me :+1:

**Pull request merged**